client/components/data/query-sites/index.jsx

```diff
@@ -15,66 +13,94 @@ import { requestSites, requestSite } from 'state/sites/actions';
-			this.requestSingle( nextProps );
-		}
+function QueryAll() {
+	const dispatch = useDispatch();
 
-		if ( nextProps.primaryAndRecent && nextProps.primarySiteId !== this.props.primarySiteId ) {
-			this.requestPrimary( nextProps );
-		}
+	useEffect( () => {
+		dispatch( requestAll() );
+	}, [ dispatch ] );
 
-		if (
-			nextProps.primaryAndRecent &&
-			! isEqual( nextProps.recentSiteIds, this.props.recentSiteIds )
-		) {
-			this.requestRecent( nextProps );
-		}
-	}
+	return null;
+}
 
-	requestAll( props ) {
-		if ( props.allSites && ! props.requestingSites ) {
-			props.requestSites();
-		}
+const requestSingle = siteId => ( dispatch, getState ) => {
+	if ( siteId && ! isRequestingSite( getState(), siteId ) ) {
+		dispatch( requestSite( siteId ) );
 	}
+};
 
-	requestPrimary( props ) {
-		if ( props.primaryAndRecent && ! props.hasAllSitesList ) {
-			const { primarySiteId, isRequestingPrimarySite } = props;
+function QuerySingle( { siteId } ) {
+	const dispatch = useDispatch();
 
-			if ( primarySiteId && ! isRequestingPrimarySite ) {
-				props.requestSite( primarySiteId );
-			}
+	useEffect( () => {
+		if ( siteId ) {
+			dispatch( requestSingle( siteId ) );
 		}
-	}
+	}, [ dispatch, siteId ] );
 
-	requestRecent( props ) {
-		if ( props.primaryAndRecent && ! props.hasAllSitesList ) {
-			const { recentSiteIds, isRequestingRecentSites, primarySiteId } = props;
+	return null;
+}
 
-			if ( recentSiteIds && recentSiteIds.length && ! isRequestingRecentSites ) {
-				forEach( without( recentSiteIds, primarySiteId ), props.requestSite );
-			}
-		}
+const requestPrimary = siteId => ( dispatch, getState ) => {
+	const state = getState();
+	if ( siteId && ! hasAllSitesList( state ) && ! isRequestingSite( state, siteId ) ) {
+		dispatch( requestSite( siteId ) );
 	}
+};
 
-	requestSingle( props ) {
-		if ( props.siteId && ! props.requestingSite ) {
-			props.requestSite( props.siteId );
-		}
+const requestRecent = siteIds => ( dispatch, getState ) => {
+	const state = getState();
+	if ( ! siteIds.length || hasAllSitesList( state ) ) {
+		return;
 	}
 
-	render() {
-		return null;
+	const isRequestingSomeSite = siteIds.some( siteId => isRequestingSite( state, siteId ) );
+
+	if ( ! isRequestingSomeSite ) {
+		return;
 	}
+
+	siteIds.forEach( siteId => dispatch( requestSite( siteId ) ) );
+};
+
+function QueryPrimaryAndRecent() {
+	const primarySiteId = useSelector( getPrimarySiteId );
+	// This should return the same reference every time, so we can compare by reference.
+	const recentSiteIds = useSelector( getRecentSites );
+	const dispatch = useDispatch();
+
+	useEffect( () => {
+		dispatch( requestPrimary( primarySiteId ) );
+	}, [ dispatch, primarySiteId ] );
+
+	useEffect( () => {
+		if ( recentSiteIds && recentSiteIds.length ) {
+			dispatch(
+				requestRecent( recentSiteIds.filter( recentSiteId => recentSiteId !== primarySiteId ) )
+			);
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why this disable? ESLint wants `primarySiteId` to be a dependency, too, so why not put it there? It looks like a valid objection.

Reply:
This is all now part of the same effect, so it was removed.